Add recipe for wpcarro/alarm.el

wpcarro opened the pull request:
Publishing alarm.el to MELPA.

### Brief summary of what the package does

Use Emacs and `ivy` to schedule alarms.

### Direct link to the package repository

https://github.com/wpcarro/alarm.el

### Your association with the package

I am the creator and maintainer of this package.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I have confirmed some of these without doing them

alphapapa commented:
I'd recommend naming the package `ivy-alarm`.  :)

walseb commented:
Since it seems like you don't use any ivy specific features you can actually convert this to use `completing-read`, making it completion framework independent. To make `completing-read` not sort the results is more complicated however, here they describe what you have to do: https://emacs.stackexchange.com/questions/41801/how-to-stop-completing-read-ivy-completing-read-from-sorting
Oh and you mention `productivity.el` in the notification title, I assume this is what the package used to be named

wpcarro commented:
Thanks @alphapapa - renamed the package `egg-timer.el`. 

@walseb thanks for the feedback. I'm using `completing-read` instead of `ivy`, so I dropped the dependency.

riscy commented:
Thanks for this.  Here's a quick pass:

### egg-timer.el
<kbd>byte-compile-file</kbd> (using Emacs 26.3):
```
egg-timer.el:78:1:Warning: the function `notifications-notify' is not known to
be defined.
```

<kbd>M-x checkdoc</kbd> (using version 0.6.1):
- No issues! 💯

<kbd>M-x package-lint-current-buffer</kbd> (using version 20191220.419):
```
2 issues found:
1:0: error: Package should have a Homepage or URL header.
66:15: error: You should depend on (emacs "25.1") if you need `alist-get'.
```

Other static checks/suggestions:
- egg-timer.el#L74: No `format` required; messages are already f-strings

<details><summary>Misc...</summary>

### Maintainer/license

- PR by wpcarro: https://github.com/wpcarro/alarm.el
- License found (through API): 'MIT License' (good)
- LICENSE excerpt: 'The MIT License (MIT)...'
- License appears to be written into individual files

### Possibly related
- https://melpa.org/#/egg
- https://melpa.org/#/named-timer
- https://melpa.org/#/timer-revert

### Summary

- (egg-timer :fetcher github :repo "wpcarro/egg-timer.el") 
- Package-Requires: emacs "24.1"
- _elisp/egg-timer.el (mit): Commonly used intervals for setting timers while working
</details>

wpcarro commented:
@riscy thanks for the feedback. I'll be pushing the changes you suggested here. Apologies for the failures with `checkdoc`, `package-lint-current-buffer`. I forgot to re-run these with my most recent commits.

I'm running `byte-compile-file` without seeing the same warnings as you are.  Any idea how I can reproduce this?

```
> emacs --version
GNU Emacs 26.3
Copyright (C) 2019 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of GNU Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.
```

wpcarro commented:
Changes pushed to upstream.

alphapapa commented:
> I'm running `byte-compile-file` without seeing the same warnings as you are. Any idea how I can reproduce this?

You're probably running it from inside your Emacs instance in which that library is already loaded.  You need to test compilation in a clean Emacs config.  You could use this: https://github.com/alphapapa/makem.sh

riscy commented:
Hi!  Sorry for so many weeks since my last check.  I notice this is still popping up:
### egg-timer.el
<kbd>byte-compile-file</kbd> (using Emacs 26.3):
```
egg-timer.el:79:1:Warning: the function `notifications-notify' is not known to
be defined.
```
You can add a `(require 'notifications)` to stop this one.

Using `notifications-notify` requires that Emacs is compiled with [dbus](https://www.gnu.org/software/emacs/manual/html_mono/dbus.html) support, so it may be worth adding that somewhere in your documentation (although Emacs will tell you if that's the case, so it's totally optional).

This package also seems relatively simple.  That's really not a bad thing, but there are also a few other [timer packages](https://melpa.org/#/?q=timer) available on MELPA.  Can you comment on the novelty of this package?  Could it be worth going to the effort of supporting users whose Emacs isn't compiled with dbus?  Could it be worth adding interactive functions that users could bind to keys, such as an interactive function that simply sets a 5-minute timer without taking the user through a prompt?

wpcarro commented:
@riscy - Thanks for the feedback. I agree with everything you've mentioned here.

- I added `(require 'notifications)`.
- Mentioned the dbus dependency in the Commentary section.
- Defined `egg-timer-do-schedule` to allow users to set precise KBDs.
- Added a section in the README illustrating how to consume `egg-timer-do-schedule`.

riscy commented:
Sorry to nag, but can you comment on the difference between https://melpa.org/#/tea-time (besides that it is about tea, instead of eggs:)? Maybe that package looks unmaintained -- or maybe you can use what you've written here to enhance it?

wpcarro commented:
> can you comment on the difference between https://melpa.org/#/tea-time (besides that it is about tea, instead of eggs

Sure. I think that would be useful documentation. I appreciate the suggestion, and I apologize that you had to remind me.

wpcarro commented:
@riscy - updated the `README.md` [here](https://github.com/wpcarro/egg-timer.el/commit/f0768e79c92aaee117020b0820b0aaac8f7810a7).

Let me know if I should add anything else. Thanks!

conao3 commented:
Are you use this package really timer for egg??
I believe this package is a Pomodoro timer.

Therefore, I think @wpcarro needs to compare it with the related packages that already exist in MELPA and explain their uniqueness.

https://melpa.org/#/?q=pomodoro
- gtk-pomodoro-indicator
- org-pomodoro
- pomidor
- pomodoro
- redtick
- tomatinho

My impression is that this package is too short and easy to define, so this code should in your personal utility file.

wpcarro commented:
> Are you use this package really timer for egg??
> I believe this package is a Pomodoro timer.
> 
> Therefore, I think @wpcarro needs to compare it with the related packages that already exist in MELPA and explain their uniqueness.

I compare this package with Pomodoro timers [here](https://github.com/wpcarro/egg-timer.el/blame/master/README.md#L48-L52).

conao3 commented:
I overlooked that section, sorry!

riscy commented:
Ok, great. :)
